plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -414,6 +420,45 @@ public RemoteIterator<LocatedFileStatus> listLocatedStatus(Path path)
         return new S3ObjectsV2RemoteIterator(listPath(path, OptionalInt.empty(), ListingMode.SHALLOW_ALL));
     }
 
+    @Override
+    public void batchDeleteFiles(Collection<Path> paths)
+            throws IOException
+    {
+        try {
+            List<Path> pathsList = paths.stream().collect(toImmutableList());
+            int maxObjectsPerRequest = 1000;
+            List<Path> currentBatch = new ArrayList<>();
+            int currentIndex = 0;
+            for (int i = 0; i < pathsList.size(); i++) {
+                if (currentIndex < maxObjectsPerRequest) {
+                    currentBatch.add(pathsList.get(i));
+                    currentIndex++;
+                }
+                else {
+                    deletePaths(currentBatch);
+                    currentIndex = 0;
+                    currentBatch = new ArrayList<>();
+                }
+            }
+            deletePaths(currentBatch);
+        }
+        catch (AmazonClientException e) {
+            throw new IOException("Exception while batch deleting paths from s3", e);
+        }
+    }
+
+    private void deletePaths(List<Path> paths)
+    {
+        if (!paths.isEmpty()) {
+            List<KeyVersion> keys = paths.stream().map(TrinoS3FileSystem::keyFromPath).map(KeyVersion::new).collect(toImmutableList());
+            DeleteObjectsRequest deleteObjectsRequest = new DeleteObjectsRequest(getBucketName(uri))
+                    .withKeys(keys)
+                    .withQuiet(false);
+
+            s3.deleteObjects(deleteObjectsRequest);
```

Comment:
Seems like we don't use `result.deletedObjects`, so we can set quite true.

(@pettyjamesm is there any kind of back pressure we need to take into consideration with this API? anything like https://github.com/trinodb/trino/issues/12516?)

Reply:
oh ok

Reply:
> (@pettyjamesm is there any kind of back pressure we need to take into consideration with this API? anything like https://github.com/trinodb/trino/issues/12516?)

None that I'm aware of, but since Presto/Trino hasn't used this API in the past and deletes aren't especially common in Athena it's possible one exists and we just don't know about it (yet). The thing to watch out for is going to be the API rate limit.

Reply:
@pettyjamesm isn't the same api limit set for `DeleteObjectRequest` ? If so then batch should send much less requests or am I missing something ?

Reply:
Fewer requests, yes- but I’m not sure whether the API rate limit actually means “client round-trips” or “deleted objects per second”. Other batch API endpoints like dynamo batch get item sometimes are just conveniences over the base API operation. The readme in [this project](https://github.com/momokatte/s3-bulk-delete) suggests that might be the case here too, but I’m not sure because I’m just working with the publicly available information on the API as well.

Reply:
ok I see, thanks. 
On the other hand with the previous model we would also try to delete as many files maybe rate would be slower. 
@findepi do you think I should  take api rate limiting into consideration ?